### ISSUE TITLE
PackageReference YamlDotNet cause error

### ISSUE THREAD

**Issue by Margoner** [dotnet/aspire#6360]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

 csproj
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <IsAspireHost>true</IsAspireHost>
    <UserSecretsId>xxxx-xxx-xxxx-xxx-xxxxx</UserSecretsId>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Aspire.Hosting.AppHost" Version="8.1.0" />
    <PackageReference Include="YamlDotNet" Version="16.1.3" />
  </ItemGroup>

</Project>
```

program.cs:
```
var builder = DistributedApplication.CreateBuilder(args);
builder.Build().Run();
```

### Expected Behavior

no error

### Steps To Reproduce

empty aspire project
only install yamldotnet
```
dotnet add package YamlDotNet
```

### Exceptions (if any)
```
fail: Aspire.Hosting.Dashboard[0]
      Failed to create resource aspire-dashboard
System.TypeInitializationException: The type initializer for 'k8s.KubernetesYaml' threw an exception.
       ---> System.TypeLoadException: Method 'ReadYaml' in type 'k8s.Models.IntOrStringYamlConverter' from assembly 'KubernetesClient, Version=14.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
         at k8s.KubernetesYaml..cctor()
         --- End of inner exception stack trace ---
         at Aspire.Hosting.Dcp.ApplicationExecutor.<>c__DisplayClass37_0.<<WatchResourceChanges>g__WatchKubernetesResourceAsync|3>d`1.MoveNext() in /_/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:line 325
         at Polly.ResiliencePipeline.<>c.<<ExecuteAsync>b__3_0>d.MoveNext()
         at Aspire.Hosting.Dcp.ApplicationExecutor.<>c__DisplayClass37_0.<<WatchResourceChanges>g__WatchKubernetesResourceAsync|3>d`1.MoveNext() in /_/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:line 325
      --- End of stack trace from previous location ---
         at Polly.Outcome`1.GetResultOrRethrow()
         at Polly.ResiliencePipeline.ExecuteAsync(Func`2 callback, CancellationToken cancellationToken)
         at Aspire.Hosting.Dcp.ApplicationExecutor.<>c__DisplayClass37_0.<<WatchResourceChanges>g__WatchKubernetesResourceAsync|3>d`1.MoveNext() in /_/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:line 325
```
### .NET Version info

.NET SDK:
 Version:           8.0.400
 Commit:            36fe6dda56
 Workload version:  8.0.400-manifests.58db758f
 MSBuild version:   17.11.3+0c8610977

 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.400\

Host:
  Version:      8.0.8
  Architecture: x64
  Commit:       08338fcaa5

.NET SDKs installed:
  8.0.400 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

### Anything else?

I realize that YamlDotNet has been referenced by 
```
    <PackageReference Include="Aspire.Hosting.AppHost" Version="8.1.0" />
```
But I believe that referencing YamlDotNet and causing errors should not be the expected behavior.

_No response_

**Comment by joperezr** [comment]
Thanks for the report @Margoner . We will look into this. Just out of curiosity, why do you need YamlDotNet reference in your AppHost project (if you can share 😃)?

**Comment by joperezr** [comment]
assigning to @davidfowl  to reach out to yamldotnet to talk about the breaking change for the method removed.

**Comment by eerhardt** [comment]
This should be fixed with `9.0.0-rc.1.24511.1`. We updated our dependency of KubernetesClient from `14.0.9` to `15.0.1`. Please update to the 9.0-rc1 version.

https://www.nuget.org/packages/Aspire.Hosting/9.0.0-rc.1.24511.1#dependencies-body-tab

![Image](https://github.com/user-attachments/assets/bcfdac5e-2378-4a59-ba0a-aba98e2a6ddf)

**Comment by eerhardt** [comment]
Another workaround if you can't update to 9.0 is to use the older `YamlDotNet`, specifically v`15.1.0`.
